crates/pindakaas/src/bool_linear.rs

```diff
@@ -1,57 +1,33 @@
 use std::{
-	cell::RefCell,
-	cmp::{max, min},
-	collections::{BTreeSet, VecDeque},
+	collections::VecDeque,
 	fmt::{self, Display},
 	iter::once,
-	ops::{Add, AddAssign, Deref, DerefMut, Mul, MulAssign, Range},
-	rc::Rc,
+	ops::{Add, AddAssign, Deref, DerefMut, Mul, MulAssign},
 };
 
-use iset::IntervalMap;
 use itertools::Itertools;
 use rustc_hash::{FxBuildHasher, FxHashMap};
 
 use crate::{
 	cardinality::Cardinality,
 	cardinality_one::{CardinalityOne, PairwiseEncoder},
 	helpers::{as_binary, is_powers_of_two, new_named_lit},
-	integer::{
-		lex_leq_const, Consistency, IntVar, IntVarEnc, IntVarOrd, Lin, Model, GROUND_BINARY_AT_LB,
-	},
+	integer::{enc::LitOrConst, lex_leq_const, IntVar},
 	propositional_logic::{Formula, TseitinEncoder},
-	sorted::{Sorted, SortedEncoder},
-	AsDynClauseDatabase, BoolVal, Checker, ClauseDatabase, ClauseDatabaseTools, Coeff, Encoder,
-	IntEncoding, Lit, Result, Unsatisfiable, Valuation,
+	AsDynClauseDatabase, BoolVal, CheckError, Checker, ClauseDatabase, Coeff, Encoder, IntEncoding,
+	Lit, Result, Unsatisfiable, Valuation,
 };
 
 #[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
 /// Encoder for the linear constraints that ∑ coeffᵢ·litsᵢ ≷ k using a
 /// binary adders circuits
 pub struct AdderEncoder {}
 
-#[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
-/// Encode the constraint that ∑ coeffᵢ·litsᵢ ≦ k using a Binary
-/// Decision Diagram (BDD)
-pub struct BddEncoder {
-	add_consistency: bool,
-	cutoff: Option<Coeff>,
-}
-
-#[derive(Debug, Clone, PartialEq)]
-/// The representation of a Binary Decision Diagram (BDD) node for the
-/// [`BddEncoder`].
-enum BddNode {
-	Val,
-	Gap,
-	View(Coeff),
-}
-
 #[derive(Clone, Debug, Default, PartialEq, Eq)]
 /// A tranformation of a general [`BoolLinear`] constraint into a aggregated and
 /// normalized variant.
 pub struct BoolLinAggregator {
-	sorted_encoder: SortedEncoder,
+	// sorted_encoder: SortedEncoder,
```

Comment:
note: i've removed the sortedencoder for now; the rewrite of Model does not support all the features the old one had, so it will take some effort to get it working again. I suggest we turn that into an issue.